src/Routing/RouteCollection.php

```diff
@@ -147,6 +147,12 @@ public function parse(string $url, string $method = ''): array
 
         if (isset($this->staticPaths[$decoded])) {
             foreach ($this->staticPaths[$decoded] as $route) {
+                // Extension parsing was enabled after the route was added to
+                // the collection and it can no longer be treated as a static route.
+                if ($route->getExtensions()) {
```

Comment:
Looks like this will never be reached, as `/foo.json` wouldn't be in the static paths array, but only `/foo`?

Reply:
Yes, this is not necessary, I will remove it. It actually prevents early parsing of `/foo` URL which would also match the same route.